### PR TITLE
Upgrade screens redo

### PR DESCRIPTION
**Pull request by DennisKiselev**
Added an upgrade app, visible only when bench model is compatible and upgrade has not already been complete. Upgrade app allows user to enter their upgrade code, and performs upgrade if code is valid.

Tested on rig, localisation checked on windows

image
image
image
image
image

### CODE REVIEW

File: src/asmcnc/apps/upgrade_app/screen_upgrade.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+            self.upgrade_and_proceed()
+        else:
+            self.show_error_message(self.l.get_str("Upgrade code incorrect, please check it and try again."))
+
+    def upgrade_and_proceed(self):
+        try:
+            self.m.enable_theateam()
+            self.sm.current = "upgrade_successful"
+        except:
+            warning_message = 'Problem creating SC2 compatability file!!'
```
Comment (LettyYeti) [inline src/asmcnc/apps/upgrade_app/screen_upgrade.py:226]: Not translated and reveals info about the unlock process onscreen, can you change to generic "Error!" and then print this error message so we can see it in logs if we need :)

File: src/asmcnc/comms/foreign_dictionary.txt [hunk 2]
```diff
@@ -709,12 +709,31 @@ SmartBench has paused the job, to prevent further damage.	SmartBench hat den Auf
...
 Enabled	Aktiviert	activé	Abilitato	Aktivert	Geactiveerd	Włączony	Aktiveret	활성화	Aktiverad	有効	Activado
 Disabled	Deaktiviert	désactivé	Disabilitato	Deaktivert	Gedeactiveerd	Wyłączony	Deaktiveret	장애가 있는	Avaktiverad	無効	Desactivado
 Failed to toggle SSH Service	SSH Dienst Umschaltung fehlgeschlagen	Impossible de commuter le service SSH	Passaggio a SSH service fallito	Kan ikke veksle SSH-tjeneste	SSH service herstart mislukt	Nie udało się przełączyć usługi SSH	Kunne ikke slå SSH-service til/fra	SSH 서비스를 전환하지 못했습니다	Misslyckade att växla till SSH service	SSHサービスの切り替えに失敗しました	Error al activar el servicio SSH
 Reboot console and try again	Starten Sie Ihre Konsole neu und versuchen Sie es erneut	Redémarrez votre console et réessayez	Riavvia la console e prova ancora	Restart konsollen og prøv på nytt	Herstart de console en probeer het opnieuw	Zrestartuj interfejs i spróbuj ponownie	Genstart din konsol og forsøg igen	콘솔을 재부팅하고 다시 시도하십시오	Starta om din manöverpanel och försök igen	コンソールを再起動し、再度お試しください。	Reinicie la consola e inténtelo de nuevo
 SSH	SSH	SSH	Shell Sicura	SSH	SSH	SSH	SSH
+Upgrade	Upgrade	mise à niveau	Aggiornamento	Päivitä	Upgrade	Modernizacja	Opgrader
```
Comment (LettyYeti) [inline src/asmcnc/comms/foreign_dictionary.txt:719]: Ensure localisation spreadsheet colours are updated to reflect additions here :)

File: src/asmcnc/skavaUI/screen_lobby.py [hunk 3]
```diff
@@ -423,6 +458,7 @@ class LobbyScreen(Screen):
     trigger_update_popup = False
     welcome_popup_description = ''
     update_message = ''
+    upgrade_app_hidden = False
```
[the remark this answers is not in the document]
Comment (DennisKiselev) [reply src/asmcnc/skavaUI/screen_lobby.py:461]: I don't think changing this variable would help, but I've changed the condition to always hide the app, can you check if it's what you're asking for?

**Review by LettyYeti — APPROVED** [review]
Looks good, just double check what the algorithm should be :)

**Pull request merged**